regress/pesterTests/Setup.Tests.ps1

```diff
@@ -0,0 +1,482 @@
+        $trustedInstallerSid = Get-UserSID -User "NT SERVICE\TrustedInstaller"
+        $allApplicationPackagesSid = Get-UserSID -User "ALL APPLICATION PACKAGES"
+        $allRestrictedApplicationPackagesSid = Get-UserSID -User "ALL RESTRICTED APPLICATION PACKAGES"
+
+        $FSReadAccessPerm = ([System.UInt32] [System.Security.AccessControl.FileSystemRights]::ReadAndExecute.value__)  -bor `
+                    ([System.UInt32] [System.Security.AccessControl.FileSystemRights]::Synchronize.value__)
+        $FSReadWriteAccessPerm = ([System.UInt32] [System.Security.AccessControl.FileSystemRights]::ReadAndExecute.value__)  -bor `
+                ([System.UInt32] [System.Security.AccessControl.FileSystemRights]::Write.value__)  -bor `
+                ([System.UInt32] [System.Security.AccessControl.FileSystemRights]::Modify.value__)  -bor `
+                ([System.UInt32] [System.Security.AccessControl.FileSystemRights]::Synchronize.value__)
+
+        $FSFullControlPerm = [System.UInt32] [System.Security.AccessControl.FileSystemRights]::FullControl.value__
+        $FSReadAndExecutePerm = ([System.UInt32] [System.Security.AccessControl.FileSystemRights]::ReadAndExecute.value__)  -bor `
+                ([System.UInt32] [System.Security.AccessControl.FileSystemRights]::Synchronize.value__)
+
+        $RegReadKeyPerm = ([System.UInt32] [System.Security.AccessControl.RegistryRights]::ReadKey.value__)
+        $RegFullControlPerm = [System.UInt32] [System.Security.AccessControl.RegistryRights]::FullControl.value__        
+
+        #only validate owner and ACEs of the registry
+        function ValidateRegistryACL {
+            param([string]$RegPath, $Ownersid = $adminsSid, $IdAcls)
+            Test-Path -Path $RegPath | Should Be $true                      
+            $myACL = Get-ACL $RegPath
+            $OwnerSid = Get-UserSid -User $myACL.Owner
+            $OwnerSid.Equals($Ownersid) | Should Be $true
+            $myACL.Access | Should Not Be $null
+            $CAPABILITY_SID = "S-1-15-3-1024-1065365936-1281604716-3511738428-1654721687-432734479-3232135806-4053264122-3456934681"            
+            $nonPropagate = $myACL.Access | ? {($_.PropagationFlags -eq ([System.Security.AccessControl.PropagationFlags]::None)) -and ($_.IdentityReference -ine $CAPABILITY_SID)}
+
+            foreach ($a in $nonPropagate) {
+                $findItem = $IdAcls | ? {
+                    ($a.IdentityReference -eq (Get-UserAccount -UserSid ($_.Identity))) -and `
+                    ($a.IsInherited -eq $_.IsInherited) -and `
+                    ($a.AccessControlType -eq ([System.Security.AccessControl.AccessControlType]::Allow)) -and  `
+                    ($a.PropagationFlags -eq ([System.Security.AccessControl.PropagationFlags]::None) -and  `
+                    (([System.Int32]$a.RegistryRights.value__) -eq ($_.RegistryRights))) 
+                }
+                $findItem | Should Not Be $null
```

Comment:
Didn't quite understand the logic here.
This foreach loop validates whether IdAcls is subset of registry ACLs? If yes, then testcase should be modified to throw an exception for any of the registry ACL not present in Idacls?

Reply:
added next line to validate each idacls can be found too.